packages/batch/src/processPartialResponse.ts

```diff
@@ -28,7 +30,12 @@ const processPartialResponse = async (
 
   const records = event['Records'];
 
-  processor.register(records, recordHandler);
+  if (options) {
+    processor.register(records, recordHandler, options);
```

Comment:
We can safely pass the options to the processor register method because the check on the options is already there, so the entire `if-block` can be simplified to:

`processor.register(event.Records, recordHandler, options);`